morpheus/stages/general/broadcast_stage.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import typing
+
+import mrc
+from mrc.core.node import Broadcast
+
+from morpheus.config import Config
+from morpheus.pipeline.stage import Stage
+from morpheus.pipeline.stream_pair import StreamPair
+
+logger = logging.getLogger(__name__)
+
+
+class BroadcastStage(Stage):
```

Comment:
So a `BroadcastStage` should only be a single port stage. It's output port is one that can be used to connect multiple downstream stages too. But they all connect to the same port. You would use it like this:

```python
pipe = Pipeline(config)
source_stage = pipe.add_stage(FileSourceStage(config, filename=input_file, iterative=False))

broadcast_stage = pipe.add_stage(BroadcastStage(config))

write_to_file_stage1 = pipe.add_stage(WriteToFileStage(config, filename=out_file1, overwrite=False))
write_to_file_stage2 = pipe.add_stage(WriteToFileStage(config, filename=out_file2, overwrite=False))

pipe.add_edge(source_stage, broadcast_stage)

pipe.add_edge(broadcast_stage, write_to_file_stage1)
pipe.add_edge(broadcast_stage, write_to_file_stage2)

pipe.run()
```
So this stage can inherit from `SinglePortStage` and doesnt need to use ports at all.